java/code/src/com/suse/manager/attestation/AttestationManager.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2024 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.suse.manager.attestation;
+
+import com.redhat.rhn.common.hibernate.LookupException;
+import com.redhat.rhn.common.security.PermissionException;
+import com.redhat.rhn.domain.action.Action;
+import com.redhat.rhn.domain.action.ActionFactory;
+import com.redhat.rhn.domain.action.CoCoAttestationAction;
+import com.redhat.rhn.domain.server.MinionServer;
+import com.redhat.rhn.domain.server.Server;
+import com.redhat.rhn.domain.user.User;
+import com.redhat.rhn.manager.action.ActionManager;
+import com.redhat.rhn.manager.system.SystemManager;
+import com.redhat.rhn.taskomatic.TaskomaticApi;
+import com.redhat.rhn.taskomatic.TaskomaticApiException;
+
+import com.suse.manager.model.attestation.AttestationFactory;
+import com.suse.manager.model.attestation.CoCoEnvironmentType;
+import com.suse.manager.model.attestation.ServerCoCoAttestationConfig;
+import com.suse.manager.model.attestation.ServerCoCoAttestationReport;
+import com.suse.manager.webui.services.pillar.MinionPillarManager;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.security.SecureRandom;
+import java.util.Base64;
+import java.util.Date;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Attestation Manager
+ */
+public class AttestationManager {
+    private static final Logger LOG = LogManager.getLogger(AttestationManager.class);
+    private final AttestationFactory factory;
+    private TaskomaticApi taskomaticApi;
+
+    /**
+     * Constructor
+     */
+    public AttestationManager() {
+        factory = new AttestationFactory();
+        taskomaticApi = new TaskomaticApi();
```

Comment:
we should take these in as parameters and configure a shared global instance like we do for others already. This way we can also get rid of the `setTaskomaticApi` and make it final.

Reply:
Not sure if we need to add this to the GlobalInstanceHolder.
We do not have TaskomaticApi there yet.

I changed this now. Please check if this is better now